# HG changeset patch
# Parent 401da9a1f4fcdf87f6105ed87757a8de4b319cde
# User Frédéric Wang <fred.wang@free.fr>
Disable support for mrow-like embellished operator to fix a regression. b=687807, r=karlt.

diff --git a/layout/mathml/nsMathMLContainerFrame.cpp b/layout/mathml/nsMathMLContainerFrame.cpp
--- a/layout/mathml/nsMathMLContainerFrame.cpp
+++ b/layout/mathml/nsMathMLContainerFrame.cpp
@@ -1496,18 +1496,21 @@ nsMathMLContainerFrame::TransmitAutomati
     // we successfully went to the end of the loop. This means that one of
     // condition 1) or 2) holds.
     if (!embellishedOpFound) {
       // the mrow-like element is space-like.
       mPresentationData.flags |= NS_MATHML_SPACE_LIKE;
     } else {
       // the mrow-like element is an embellished operator.
       // let the state of the embellished operator found bubble to us.
-      mPresentationData.baseFrame = baseFrame;
-      mEmbellishData = embellishData;
+      // XXXfred: Disable support for mrow-like embellished operator.
+      // Temporarily solution to avoid a regression (see bug 687807).
+      embellishedOpFound = false;
+      // mPresentationData.baseFrame = baseFrame;
+      // mEmbellishData = embellishData;
     }
   }
 
   if (childFrame || !embellishedOpFound) {
     // The element is not embellished operator
     mPresentationData.baseFrame = nullptr;
     mEmbellishData.flags = 0;
     mEmbellishData.coreFrame = nullptr;
diff --git a/layout/reftests/mathml/reftest.list b/layout/reftests/mathml/reftest.list
--- a/layout/reftests/mathml/reftest.list
+++ b/layout/reftests/mathml/reftest.list
@@ -31,39 +31,41 @@ fails-if(smallScreen&&Android) fuzzy(255
 == mfenced-9.html mfenced-9-ref.html
 fails-if(winWidget) == mfenced-10.html mfenced-10-ref.html
 == mi-mathvariant-1.xhtml mi-mathvariant-1-ref.xhtml
 == mi-mathvariant-2.xhtml mi-mathvariant-2-ref.xhtml
 != non-spacing-accent-1.xhtml non-spacing-accent-1-ref.xhtml
 == overbar-width-1.xhtml overbar-width-1-ref.xhtml
 == quotes-1.xhtml quotes-1-ref.xhtml
 != stretchy-underbar-1.xhtml stretchy-underbar-1-ref.xhtml 
+!= stretchy-munderover-1.html stretchy-munderover-1-ref.html
 == table-width-1.xhtml table-width-1-ref.xhtml
 == table-width-2.html table-width-2-ref.html
 == underbar-width-1.xhtml underbar-width-1-ref.xhtml
 == mathml-type-supported.xhtml mathml-type-supported-ref.xml
 == mtable-align-negative-rownumber.html mtable-align-negative-rownumber-ref.html
++# XXXfredw Support for mrow-like embellished op is disabled. See bug 687807.
 != embellished-op-1-1.html embellished-op-1-1-ref.html
-!= embellished-op-1-2.html embellished-op-1-2-ref.html
-!= embellished-op-1-3.html embellished-op-1-3-ref.html
-!= embellished-op-1-4.html embellished-op-1-4-ref.html
-!= embellished-op-1-5.html embellished-op-1-5-ref.html
-!= embellished-op-2-1.html embellished-op-2-1-ref.html
-!= embellished-op-2-2.html embellished-op-2-2-ref.html
-!= embellished-op-2-3.html embellished-op-2-3-ref.html
-!= embellished-op-2-4.html embellished-op-2-4-ref.html
-!= embellished-op-3-1.html embellished-op-3-1-ref.html
-!= embellished-op-3-2.html embellished-op-3-2-ref.html
-!= embellished-op-3-3.html embellished-op-3-3-ref.html
-!= embellished-op-3-4.html embellished-op-3-4-ref.html
-!= embellished-op-3-5.html embellished-op-3-5-ref.html
+fails != embellished-op-1-2.html embellished-op-1-2-ref.html
+fails != embellished-op-1-3.html embellished-op-1-3-ref.html
+fails != embellished-op-1-4.html embellished-op-1-4-ref.html
+fails != embellished-op-1-5.html embellished-op-1-5-ref.html
+fails != embellished-op-2-1.html embellished-op-2-1-ref.html
+fails != embellished-op-2-2.html embellished-op-2-2-ref.html
+fails != embellished-op-2-3.html embellished-op-2-3-ref.html
+fails != embellished-op-2-4.html embellished-op-2-4-ref.html
+fails != embellished-op-3-1.html embellished-op-3-1-ref.html
+fails != embellished-op-3-2.html embellished-op-3-2-ref.html
+fails != embellished-op-3-3.html embellished-op-3-3-ref.html
+fails != embellished-op-3-4.html embellished-op-3-4-ref.html
+fails != embellished-op-3-5.html embellished-op-3-5-ref.html
 == embellished-op-4-1.html embellished-op-4-1-ref.html
 == embellished-op-4-2.html embellished-op-4-2-ref.html
 == embellished-op-4-3.html embellished-op-4-3-ref.html
-== embellished-op-5-1.html embellished-op-5-ref.html
+fails == embellished-op-5-1.html embellished-op-5-ref.html
 == embellished-op-5-2.html embellished-op-5-ref.html
 == semantics-1.xhtml semantics-1-ref.xhtml
 == semantics-2.html semantics-2-ref.html
 == semantics-3.html semantics-3-ref.html
 != mathcolor-1.xml mathcolor-1-ref.xml
 != mathcolor-2.xml mathcolor-2-ref.xml
 != mathcolor-3.xml mathcolor-3-ref.xml
 == mathcolor-4.xml mathcolor-4-ref.xml
diff --git a/layout/reftests/mathml/stretchy-munderover-1-ref.html b/layout/reftests/mathml/stretchy-munderover-1-ref.html
new file mode 100644
--- /dev/null
+++ b/layout/reftests/mathml/stretchy-munderover-1-ref.html
@@ -0,0 +1,19 @@
+<!doctype>
+<html>
+  <head>
+    <title>Stretchy munderover</title>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+  </head>
+  <body>
+    <!-- See bug 687807 -->
+    <math>
+      <mrow>
+	<munderover>
+	  <mo stretchy="true">&#x2192;</mo>
+          <mtext>AAAAAA</mtext>
+          <mtext>BBBBBB</mtext>
+	</munderover>
+      </mrow>
+    </math>
+  </body>
+</html>
diff --git a/layout/reftests/mathml/stretchy-munderover-1.html b/layout/reftests/mathml/stretchy-munderover-1.html
new file mode 100644
--- /dev/null
+++ b/layout/reftests/mathml/stretchy-munderover-1.html
@@ -0,0 +1,19 @@
+<!doctype>
+<html>
+  <head>
+    <title>Stretchy munderover</title>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+  </head>
+  <body>
+    <!-- See bug 687807 -->
+    <math>
+      <mrow>
+	<munderover>
+	  <mo stretchy="false">&#x2192;</mo>
+          <mtext>AAAAAA</mtext>
+          <mtext>BBBBBB</mtext>
+	</munderover>
+      </mrow>
+    </math>
+  </body>
+</html>
